src/mercury_engine_data_structures/type_lib.py

```diff
@@ -171,12 +167,6 @@ def _find_type_errors(self, __value: typing.Any) -> BaseException | None:
                 return None
             return ValueError(f"{__value} is out of range of [{hex(low)}, {hex(high)}]")
 
-        if self.primitive_kind in primitive_vector_lengths:
-            length = primitive_vector_lengths[self.primitive_kind]
-            if len(__value) == length and all(isinstance(v, float) for v in __value):
-                return None
-            return ValueError(f"Invalid CVector{length}D: {__value}")
-
         return None
```

Comment:
you should keep this length check, since `isinstance(Vec4(0.0, 0.0, 0.0, 0.0), Vec2)` is true. the float check isn't strictly necessary but doesn't hurt

Reply:
Do we want that to explicitly fail? Shouldn't you be using `type(Vec4(0.0, 0.0, 0.0, 0.0)) is Vec2` if you want to exclude subclasses?

Reply:
you could you use that instead of this length check if you prefer. still needs to be a special case tho